pkg/subctl/cmd/validate_fw_vxlan.go

```diff
@@ -59,47 +61,56 @@ func validateFirewallVxLANConfig(cmd *cobra.Command, args []string) {
 		}
 
 		status.End(cli.Success)
-		validateVxLANConfigWithinCluster(item.config, item.clusterName, submariner)
+		if !validateVxLANConfigWithinCluster(item.config, item.clusterName, submariner) {
+			validationStatus = false
+		}
+	}
+
+	if !validationStatus {
+		exitWithErrorMsg("Error occur while trying to Check firewall access for Submariner VXLAN traffic")
 	}
 }
 
-func validateVxLANConfigWithinCluster(config *rest.Config, clusterName string, submariner *v1alpha1.Submariner) {
+func validateVxLANConfigWithinCluster(config *rest.Config, clusterName string, submariner *v1alpha1.Submariner) bool {
 	status.Start(fmt.Sprintf("Checking the firewall configuration to determine if VXLAN traffic is allowed"+
 		" in cluster %q", clusterName))
-	validateFWConfigWithinCluster(config, submariner)
+	validationStatus := validateFWConfigWithinCluster(config, submariner)
 	status.End(status.ResultFromMessages())
+	return validationStatus
 }
 
-func validateFWConfigWithinCluster(config *rest.Config, submariner *v1alpha1.Submariner) {
+func validateFWConfigWithinCluster(config *rest.Config, submariner *v1alpha1.Submariner) bool {
 	if submariner.Status.NetworkPlugin == "OVNKubernetes" {
 		status.QueueSuccessMessage("This check is not necessary for the OVNKubernetes CNI plugin")
-		return
+		return true
 	}
 
 	clientSet, err := kubernetes.NewForConfig(config)
 	if err != nil {
 		message := fmt.Sprintf("Error creating API server client: %s", err)
 		status.QueueFailureMessage(message)
-		return
+		return false
 	}
 
 	gateways := getGatewaysResource(config)
 	if gateways == nil {
-		status.QueueWarningMessage("There are no gateways detected on the cluster.")
-		return
+		message := "There are no gateways detected on the cluster."
+		status.QueueWarningMessage(message)
```

Comment:
nit but why change this to store message in a var?

```suggestion
		status.QueueWarningMessage("There are no gateways detected on the cluster.")
```